message/validation/signer_state.go

```diff
@@ -22,6 +22,9 @@ type SignerState struct {
 	// 10 (f=3): C(10,7)+C(10,8)+C(10,9)+C(10,10)=176
 	// 13 (f=4): C(13,9)+C(13,10)+C(13,11)+C(13,12)+C(13,13)=1093
 	SeenSigners map[SignersBitMask]struct{}
+	// MaxRCJ stores max seen round change justification length.
+	// It always increases, making same round messages with shorter RCJ ignored.
+	MaxRCJ uint8
```

Comment:
Consider adding a `Reset()` method comment explaining why `MaxRCJ` is intentionally NOT reset. This prevents future confusion or "fixes" that could introduce security vulnerabilities.

Suggested update:
```go
// Reset resets the state's round, message counts, and proposal data to the given values.
// It also updates the start time to the current time.
// Note: MaxRCJ is intentionally NOT reset to maintain security across rounds within the same slot.
// This prevents accepting round change messages with weaker justifications in later rounds.
func (s *SignerState) Reset(slot phase0.Slot, round specqbft.Round) {